test/core/ChainManager.test.js

```diff
@@ -0,0 +1,131 @@
+import SocketService from '../../src/core/SocketService';
+import ChainManager from '../../src/core/ChainManager';
+import {
+  listAllChains,
+  listAllSnapshots,
+  getBlockNumber,
+  mineBlock
+} from '../../src/core/ChainRequest';
+
+jest.setTimeout(10000);
+let socket, service, id, chain, exists, response, block;
+
+const options = {
+  accounts: 3,
+  block_mine_time: 0,
+  clean_on_stop: false
+};
+
+beforeEach(async () => {
+  socket = new SocketService();
+  service = new ChainManager(socket);
+  await socket.init();
+  await service.init();
+  id = await service.createChain({ ...options });
+});
+
+afterEach(async () => {
+  await service.clean();
+});
+
+test('service will create chain instance', async () => {
+  chain = service.chain(id);
+  expect(chain.id).toEqual(id);
+  expect(chain.name).toEqual(`chain:${id}`);
+  expect(chain.active).toBe(true);
+  expect(chain.accounts.length).toBe(options.accounts + 1);
+});
+
+test('service will stop chain instance', async () => {
+  expect(service.chain(id).active).toBe(true);
+  await service.chain(id).stop();
+  expect(service.chain(id).active).toBe(false);
+});
+
+test('service will restart chain instance', async () => {
+  await service.chain(id).stop();
+  expect(service.chain(id).active).toBe(false);
+  await service.chain(id).start();
+  chain = service.chain(id);
+  expect(chain.active).toBe(true);
+  expect(chain.accounts.length).toBe(options.accounts + 1);
+});
+
+test('service will check existence of chain', async () => {
+  exists = await service.exists(id);
+  expect(exists).toBe(true);
+  id = new Array(20).join('1');
```

Comment:
is there some other test that depends upon this `id` being an incorrect value? if not, we should not be re-assigning the value of this global variable again. better to be explicit and clear about what you are and aren't doing by saying instead
```js
const badId = new Array(20).join('1');
```